lxd/certificates.go

```diff
@@ -477,7 +477,28 @@ func certificatesPost(d *Daemon, r *http.Request) response.Response {
 	}
 
 	// If caller is already trusted and does not have permission to create certificates, they cannot create more certificates.
-	if trusted && !userCanCreateCertificates && req.Certificate == "" && !req.Token {
+	// If trust token is provided, also return an error because trust token can only be used to add a remote.
+	if trusted && (req.Certificate == "" && !req.Token && !userCanCreateCertificates || req.TrustToken != "") {
```

Comment:
I'm uncertain which edge cases the changes made here are catching.

Can we just have `if trusted and req.TrustToken != ""`?

Nobody, regardless of their permissions, should be sending a trust token if they are already trusted right?

Reply:
It is `if trusted and req.TrustToken != ""` - I just left the previous condition `trusted && (req.Certificate == "" && !req.Token && !userCanCreateCertificates` there to not mess with anything else (and not duplicate the error).

So it can be read as if `trusted && req.TrustToken != ""` or `trusted && req.Certificate == "" && !req.Token && !userCanCreateCertificates`.

Should I split it?